.github/workflows/tests.yml

```diff
@@ -8,38 +8,42 @@ on:
   pull_request:
 
 jobs:
-  unit-integration:
+  unit-and-integration:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
 
-      - name: Setup Go
-        uses: actions/setup-go@v5
-        with:
-          go-version: ">=1.21.0"
+      - name: Run unit tests
+        run: make test-unit
 
-      - name: Run unit & integration tests
-        run: make test
-  e2e-quickstart:
+      - name: Run integration tests
+        run: make test-integration
+
+  e2e:
     runs-on: ubuntu-latest
+    needs: unit-and-integration # No use in running e2e tests if integration tests fail.
```

Comment:
I would prefer to remove this needs

Reply:
if we start running out of resources, then we can consider adding it back.

Reply:
done